File: occasum-core/src/main/java/pw/stamina/occasum/properties/collection/CollectionPropertyAdapter.java
package pw.stamina.occasum.properties.collection;

import pw.stamina.occasum.properties.PropertyParseException;

import java.util.Collection;

public interface CollectionPropertyAdapter<T> {

    Collection<T> parse(String input) throws PropertyParseException;

    String toString(Collection<T> elements);
}

File: occasum-core/src/main/kotlin/pw/stamina/occasum/properties/collection/CollectionPropertyAdapter.kt
package pw.stamina.occasum.properties.collection

import pw.stamina.occasum.properties.PropertyParseException

interface CollectionPropertyAdapter<T> {

    @Throws(PropertyParseException::class)
    fun parse(input: String): Collection<T>

    fun toString(elements: Collection<T>): String
}
